### DISCUSSION TITLE
Alternative approaches to variation partitioning with correlated predictors

### DISCUSSION THREAD

**Discussion by RodolfoPelinson** [JenniNiku/gllvm#203] [Q&A]
First, thanks for all of the amazing work put in the gllvm package!

I have a situation where I have two groups of predictors that are reasonably correlated. I would like to know how much of the variance explained by these predictors is shared and how much is not, something like a regular variation partitioning done with R² values in partial redundancy analysis (pRDA). Is there a way to do something similar in the gllvm framework?

I tried to calculate the likelihood ratio pseudo R² using "null" and "full" gllvm models and did the partitioning by hand. It all seemed to work fine and the values make sense, though I don't know if this procedure makes sense in the gllvm context.

I have used the varPartitioning() function, but it does not give a "shared" fraction. The help vignette even has a warning of "In case of correlated predictors, it is advised to group them into a same group". I wonder what the consequence would be of having correlated predictors in different groups in this case. Maybe the "best" predictor would explain most of the variance, leaving not much to be explained by other correlated predictors?

I would really appreciate any thoughts on this matter.

Thanks.

**Comment by BertvanderVeen** [comment]
Thanks Rodolfo. Admittedly, I do not think about variation partitioning a lot, and @JenniNiku might be much more adept at answering your question.

The straightforward answer is: variation partitioning in the gllvm framework is presently much harder than when dealing with more traditional ordination methods, including RDA. This is because not all dimensions are estimated, and thus the total variation on the link scale is unknown, and only therelative contribution of effects as part of the model can easily be quantified. 

That is not to say all is lost - things just work a bit differently, and are potentially a bit harder to work out. 

As a potential avenue you could look into R squared and how such things are done in mixed models, eg https://stats.stackexchange.com/questions/7240/proportion-of-explained-variance-in-a-mixed-effects-model and https://pmc.ncbi.nlm.nih.gov/articles/PMC8162244/ are good places to start.

In my experience, this gets you going down a rabbit hole as R squared metrics are difficult to define for non-normal responses and those that exist wildly vary in their performance (from a population perspective where there is a "true R squared" that you are trying to estimate, if such a thing exists).
